### ISSUE TITLE
`export default` doesn't work when file contains multiple named exports

### ISSUE THREAD

**Issue by mayank99** [11ty/eleventy#3288]
### Operating system

macOS 14.5

### Eleventy

3.0.0-alpha.10

### Describe the bug

In a `.11ty.js` file, you can `export default` stuff, like this:

```js
export default () => "<h1>hello</h1>";
```

But as soon as you export anything else from the file, the above code stops working.

```js
export const foo = "test";

export default () => "<h1>hello</h1>";
```

This becomes a problem when something like `data` needs to be exported from the file.

### Expected behavior

Adding a named export should not break an existing template containing `export default`.

As a workaround, `export const render` can be used in the meantime.

```diff
  export const foo = "test";
  export const data = {…};

- export default () => "<h1>hello</h1>";
+ export const render = () => "<h1>hello</h1>";
```

But it would be nice if `export default` continued to work regardless of what else is exported from the file.

### Reproduction URL

https://stackblitz.com/edit/11ty-wtohfm?file=index.11ty.js

### Additional comments

I was looking at the code and found that this might be deliberately coded like this, though I'm not sure of the reason (maybe it had something to do with CJS?)

https://github.com/11ty/eleventy/blob/d30b952ad59e8978b22616797287fbb0fbfb332f/src/Util/Require.js#L76-L79

**Comment by zachleat** [comment]
Hey! That last one comment should be limited to objects that only have `default`. The real culprit is here: https://github.com/11ty/eleventy/blob/29f217477285dfc8d068a6b1f0d51f82cc24d06b/src/Engines/JavaScript.js#L42

**Comment by zachleat** [comment]
Fix shipping with 3.0.0-alpha.11
